portfolio/src/main/webapp/map.js

```diff
@@ -17,11 +17,144 @@
+      'The Metropolitan Museum of Art in NYC.');
+  addImageLandmark(
+      map, 51.5033273, -0.1217317, 'London Eye',
+      'The London Eye by the Thames.');
+  addImageLandmark(
+      map, 51.514510, -0.098420, 'St. Pauls Cathedral',
+      'St Paul\'s Cathedral in London.');
+  addImageLandmark(
+      map, 50.6661297, -4.7590606, 'Tintagel, Cornwall',
+      'The coast at Tintagel, Cornwall');
+  addImageLandmark(
+      map, 51.752776, -1.253692, 'Oxford, England',
+      'The view of Oxford from atop the University Church ' + 
+      'of St Mary the Virgin.');
+  addImageLandmark(
+      map, 51.1462534, -2.7157123, 'King Arthur\s Tomb',
+      'Glastonbury Abbey by King Arthur\'s Tomb.');
+  addImageLandmark(
+      map, 48.8594193, 2.3004549, 'Eiffel Tower',
+      'A side street near the Eiffel Tower.');
+  addImageLandmark(
+      map, 48.8552017, 2.3150708, 'Rodin Museum',
+      'The Thinker Statue at the Rodin Museum in Paris.');
+  addImageLandmark(
+      map, 48.8322871, 2.3846624, 'Paris Shops',
+      'Outdoor shop in Paris.');
+  addImageLandmark(
+    map, 41.2465819,1.8223518, 'Sitges, Spain',
+    'Near our stay in Sitges, Spain.');
+  addImageLandmark(
+    map, 52.3668475, 4.891357, 'Floating Flower Market',
+    'The floating flower market in Amsterdam.');
+  addImageLandmark(
+    map, 52.3773409, 4.8987192, 'Basilica in Amsterdam',
+    'By the canals with a view of the Basilica in Central Amsterdam.');
+
+  // add additional hardcoded markers with places I want to visit someday
+  addPlaceMarker(
+    map, 31.6003468, -8.0824187, 'Marrakesh',
+    'Though my last spring break trip to Morocco was cancelled, ' +
+    'I hope I can go again next spring.');
+  addPlaceMarker(
+    map, -13.1632509, -72.5452093, 'Machu Picchu',
+    'Seeing Machu Picchu is on my bucket list. ' + 
+    'Visiting Peru would be so cool!');
+  addPlaceMarker(
+    map, -8.5525233, 114.8082374, 'Bali, Indonesia',
+    'I would love to visit Bali one day!');
+  addPlaceMarker(
+    map, 26.4474128, 80.198301, 'Kanpur, India',
+    'My extended family lives in India - I haven\'t visited in a while!');
+  addPlaceMarker(
+    map, 37.4220041, -122.0862462, 'Googleplex Mountain View',
+    'I\'ve never seen Google\'s Mountain View HQ before.');
+  addPlaceMarker(
+    map, 40.4571004, -79.9171171, 'Google Pittsburgh Office',
+    'My current team is based in Pittsburgh. I\'d love to visit Google PIT!');
+
+  // create and display legend of icons
+  let legend = document.getElementById('legend');
+  for (var key in icons) {
+    let type = icons[key];
+    let name = type.name;
+    let icon = type.icon;
+    let div = document.createElement('div');
+    div.innerHTML = '<img src="' + icon + '"> ' + name;
+    legend.appendChild(div);
+  }
+  map.controls[google.maps.ControlPosition.RIGHT_BOTTOM].push(legend);
+}
+
+/** 
+ * Adds a marker that shows an info window when clicked.
+ * Represents a place pictured in the image gallery.
+ */
```

Comment:
Now that you've got methods with lots of parameters of unclear type, it's reasonable to request JSDocs. (not going to require it for all methods you write, but where you think it's reasonable that somebody might not know what type of object a map is).
https://google.github.io/styleguide/jsguide.html#jsdoc

```javascript
/** 
 * Adds a marker that shows an info window when clicked.
 * Represents a place pictured in the image gallery.
 * @param {type} paramName A brief description of this parameter.
 */
```
Since map is the tricky one, I believe you'd put something like:

```javascript
/**
 * ...
 * @param {google.maps.Map} map Manages the state of the visual Google Maps element on the page.
 * ...
 */
```